isoCluster.py

```diff
@@ -0,0 +1,279 @@
+import sys
+import os
+from pathlib import Path
+import shutil
+import ipaddress
+from dataclasses import dataclass
+import re
+import time
+import urllib.parse
+from concurrent.futures import Future
+from concurrent.futures import ThreadPoolExecutor
+from typing import Dict
+from typing import List
+from typing import Tuple
+from logger import logger
+from clustersConfig import ClustersConfig
+from clustersConfig import NodeConfig
+from clustersConfig import ExtraConfigArgs
+import host
+import common
+
+
+"""
+ExtraConfigIPU is used to provision and IPUs specified via Redfish through the IMC.
+This works by making some assumptions about the current state of the IPU:
+- The IMC is on MeV 1.2 / Mev 1.3
+- BMD_CONF has been set to allow for iso Boot
+- ISCSI attempt has been added to allow for booting into the installed media
+- The specified ISO contains full installation kickstart / kargs required for automated boot
+- The specified ISO handles installing dependencies like dhclient and microshift
+- The specified ISO architecture is aarch64
+- Password authentication has been enabled on IMC (required to be able to connect via paramiko)
+- There is an additional connection between the provisioning host and the acc on an isolated subnet to serve dhcp / provide acc with www
+"""
+
+
+@dataclass
+class DhcpdSubnetConfig:
+    subnet: str
+    netmask: str
+    range_start: str
+    range_end: str
+    broadcast_address: str
+    routers: str
+    dns_servers: list[str]
+
+
+@dataclass
+class DhcpdHostConfig:
+    hostname: str
+    hardware_ethernet: str
+    fixed_address: str
+
+
+def get_subnet_ip(ipv4_address: str, subnet_mask: str) -> str:
+    subnet_mask_bits = subnet_mask.split('.')
+    prefix_length = sum(bin(int(octet)).count('1') for octet in subnet_mask_bits)
+    cidr_network = f"{ipv4_address}/{prefix_length}"
+    network = ipaddress.ip_network(cidr_network, strict=False).network_address
+    return str(network)
```

Comment:
I think, such general utility functions belong to common.py (or similar).

Also, as there will be several functions that do something with IP addresses (in string form), their name should all share a common (but unique) prefix. For example, on th/dnsmasq branch I would add `ipaddr_norm()` instead of `normalize_ip()` ("ipaddr_" is so far not used for other topics).